enhancements/results-forwarding.md

```diff
@@ -0,0 +1,513 @@
+
+The Compliance Operator also stores raw results (Asset Reporting Format or ARF)
+in a persistent volume claim by default. While this is useful, not everyone has
+access to persistent storage. This issue is apparent in shared testing clusters
+where automation attempts to delete the Compliance Operator namespace, only to
+have the request fail because of a failed volume allocation.
+
+#### Gathering Evidence
+
+ARF files contain a lot of information about the state of the system where the
+compliance scan was effectuated. We could take the view that this is "evidence"
+gathered for the system in check. With this in mind, we can generalize and
+expand the concept to allow for forwarding this evidence to an alternative
+object store. Object storage would be a good example.
+
+In the future, we could expand the "evidence" gathered, e.g. for Kubernetes
+compliance scans we could simply compress the gathered objects the
+`api-resource-collector` downloads and send that too.
+
+While evidence is important to consider, we're going to dedicated a separate
+[enhancement](https://github.com/ComplianceAsCode/compliance-operator/pull/66)
+to that problem. This particular enhancement is focused on forwarding results.
+
+### Goals
+
+* Provide a generic way to implement compliance result forwarding for
+  Compliance Operator. This should be flexible enough to accommodate the
+  current implementation (CRDs and PVCs), while allowing for custom forwarding
+  implementations.
+
+* Define and implement a stable API for forwarder implementations.
+
+* Implement an interface for forwarding result CRDs.
+
+* Implement an interface for forwarding remediation CRDs.
+
+* Implement a switch to disable in-cluster storage if forwarding is enabled and healthy.
+
+### Non-Goals
+
+* Deprecate the current CRD/PVC approach: There is still a use-case for this.
+
+* Expand the "evidence" that's currently gathered by the Compliance Operator. This is
+  needed but should be done as part of a separate initiative.
+
+* Implement an archiving solution. That's being discussed as a separate
+  [enhancement](https://github.com/ComplianceAsCode/compliance-operator/pull/66).
+
+## Proposal
+
+The Compliance Operator will forward compliance results and remediation
+recommendations if the supplied forwarding endpoint is valid. The Compliance
+Operator will also continue to use CRDs and PVCs to store results,
+remediations, and evidence as it does today.
+
+A user may explicitly disable in-cluster storage if forwarding is enabled. The
+Compliance Operator will not support disabling PVC storage without a valid
+forwarding endpoint.
+
+This will require changing the **aggregator** to always forward via a gRPC API
+to a configured implementation. In this case, the **aggregator** will act as a
+gRPC client. The **resultserver** then becomes a gRPC server, subject to the
+evidence forwarding provider and will be renamed **evidence-persistor** and
```

Comment:
This might be partially leftover copy, but since this is the "result" proposal, describing changes to the resultserver doesn't seem necessary, since that is related to evidence/raw result. 
I'd prefer that we be precise about exactly which objects we want to forward - I think all we would forward with this proposal is ComplianceCheckResults and ComplianceRemediations and those are handled by the aggregator.

Reply:
In that case, you're advocating for keeping the results server the same but keeping the gRPC implementation in the aggregator?

Reply:
Yes, at least that is my first impression when looking over the code. But, generally I think this is a detail that will be hashed out fully when we try to implement it..